### PR TITLE
Support parsing methods as well

### PR DESCRIPTION
**Pull request by valscion**
Oh well, I guess I really wanna go head first to tackle #2 :smile:

This PR starts off with a proof of concept to see if I can get any output for method parsing. Turns out there doesn't seem to be much left to do to get some output that is actual working code

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ function parseLinks(description/*: Cheerio*/) {
  *
  * @param {Store} store
  */
-async function requestAndParse(store/*: Store*/) {
+async function requestAndParse(store/*: Store*/, hacketyhack/*: 'Field' | 'Parameters' */) {
```
Comment (sergeysova) [inline lib/parser.js]: 😆

**Comment by valscion** [comment]
Any ideas on how to progress with this? If you feel like you have ideas, now that you have invited me to contribute I can also push this branch to this repository and let you figure things out as well :)

I'm mostly interested in getting this feature supported and am not looking for glory, so I don't mind if you come up with a better idea :smile:

File: lib/parser.js (hunk 1 above)
Comment (valscion) [reply lib/parser.js]: Yeah I just tried to get the parser going :smile:

This was funky to run on the API website in developer tools console:

```js
Array.from(document.body.querySelectorAll('body table')).map(table => [table, table.querySelector(':scope tr:first-child td:first-child')]).filter(([_, el]) => el.innerText === 'Parameters').map(([table, el]) => {

let elTry = table;
for (let i = 0; i < 50; i++) {
  elTry = elTry.previousElementSibling;
  if (elTry === null) { return null; }
  if (elTry.nodeName === 'H4') { return elTry; } 
}

}).map(heading => heading.innerText)
```

I wanted to see if the same approach of finding names would work, and seems like it does :smile:

File: javascript/index.js.flow [hunk 2]
```diff
@@ -11,6 +11,36 @@ declare module "telegram-typings" {
...
+
+  /**
+   * TODO parse union properly
+   */
+  declare type Integer or String = Integer | String;
+
+  /**
+   * TODO parse union properly
+   */
+  declare type InlineKeyboardMarkup or ReplyKeyboardMarkup or ReplyKeyboardRemove or ForceReply = InlineKeyboardMarkup | ReplyKeyboardMarkup | ReplyKeyboardRemove | ForceReply;
```
Comment (valscion) [inline javascript/index.js.flow]: Oh yeah my hack didn't touch these declarations at all. Only their references 😝

File: javascript/index.js.flow [hunk 3]
```diff
@@ -3111,4 +3141,1732 @@ declare module "telegram-typings" {
...
+
+  /**
+   * Use this method to send text messages. On success, the sent Message is returned.
+   * @see https://core.telegram.org/bots/api#message
+   */
+  declare type sendMessage = {
+    /**
+     * Yes
+     */
+    chat_id: IntegerOrString,
```
Comment (valscion) [inline javascript/index.js.flow]: Such pretty type name. I guess we will really want to have `chat_id: number | string` here? As `Integer` is a native type of `number` and `String` is a native type of `string`.

File: javascript/index.js.flow [hunk 4]
```diff
@@ -3111,4 +3141,1732 @@ declare module "telegram-typings" {
...
+
+    /**
+     * Optional
+     */
+    reply_to_message_id: number,
+
+    /**
+     * Optional
+     */
+    reply_markup: InlineKeyboardMarkupOrReplyKeyboardMarkupOrReplyKeyboardRemoveOrForceReply,
```
Comment (valscion) [inline javascript/index.js.flow]: And here we will want to have:

```js
reply_markup: InlineKeyboardMarkup | ReplyKeyboardMarkup | ReplyKeyboardRemove | ForceReply
```

As the description says "Optional", should this then be 

```js
reply_markup?: InlineKeyboardMarkup | ReplyKeyboardMarkup | ReplyKeyboardRemove | ForceReply
```

right?

File: javascript/index.js.flow (hunk 3 above)
Comment (sergeysova) [reply javascript/index.js.flow]: In this case, I agree

File: javascript/index.js.flow (hunk 4 above)
Comment (sergeysova) [reply javascript/index.js.flow]: Yes

**Comment by valscion** [comment]
Closing in favor of #18

**Pull request closed without merging**